tests/main/views/test_direct_award.py

```diff
@@ -326,3 +327,38 @@ def test_search_api_urls_convert_to_correct_frontend_urls(self, search_api_url,
         doc = html.fromstring(body)
         assert html_escape(frontend_url) in body
         assert len(doc.xpath('//a[@href="{}"]'.format(frontend_url)))
+
+
+class TestDirectAwardEndSearch(TestDirectAwardBase):
+    def setup_method(self, method):
+        super(TestDirectAwardEndSearch, self).setup_method(method)
+
+        self.g9_search_results = self._get_g9_search_results_fixture_data()
+
+        search_api_client.search_services = mock.Mock()
+        search_api_client.search_services.return_value = self.g9_search_results
+
+        data_api_client.get_framework = mock.Mock()
+        data_api_client.get_framework.return_value = self._get_framework_fixture_data('g-cloud-9')
+
+        data_api_client.get_direct_award_project = mock.Mock()
+        data_api_client.get_direct_award_project.return_value = self._get_direct_award_project_fixture()
+
+        data_api_client.find_direct_award_project_searches = mock.Mock()
+        data_api_client.find_direct_award_project_searches.return_value = \
+            self._get_direct_award_project_searches_fixture()
+
+    def test_end_search_page_renders(self):
+        self.login_as_buyer()
+        res = self.client.get('/buyers/direct-award/g-cloud/projects/1/end-search')
+        assert res.status_code == 200
+        end_search_found = re.findall(r'<h1>\s*End search\s*<\/h1>', res.get_data(as_text=True))
```

Comment:
I think we generally use xpath as opposed to text/regex matches now, so it would be nice to use that if you can. Should pretty much just be as below (albeit structured more nicely)
```
from lxml import html
assert len(html.fromstring(res.get_data(as_text=True).xpath('//h1[text()="End search"]') == 1
```

Reply:
In my head the rule is that if you want to test the structure, it's better to use xpath. In this case we're testing that an H1 is there so you can definitely argue that xpath is better, although I'm not too fussed personally ;)

Reply:
I assume you mean 'xpath _isn't better', but I'm ok with you leaving it as-is. ;)